EngineLayer/ClassicSearch/ClassicSearchEngine.cs

```diff
@@ -167,6 +184,34 @@ protected override MetaMorpheusEngineResults RunSpecific()
             return new MetaMorpheusEngineResults(this);
         }
 
+        private void AddPeptideCandidateToPsm(ScanWithIndexAndNotchInfo scan, object[] myLocks, double thisScore, PeptideWithSetModifications peptide, List<MatchedFragmentIon> matchedIons)
```

Comment:
I'm wary of performance issues here because of the frequency with which this method is called. Can you benchmark the search time needed before and after this PR using the vignette?

Reply:
before this PR, the search time is 00:03:32.9719787
after this PR, the search time is 00:03:35.0665066.
The data I used is CAST_frac4&&frac5